xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-test/xwiki-platform-notifications-test-docker/src/test/it/org/xwiki/platform/notifications/test/ui/NotificationsIT.java

```diff
@@ -173,6 +174,19 @@ void simpleNotifications(TestUtils setup, TestReference testReference) throws Ex
         }
         setup.createPage(space, "DTP", "Deletion test page", "Deletion test content");
 
+        // Switch to superadmin to wait for the events queues to be fully processed before checking the notification's display with user 2.
+        setup.loginAsSuperAdmin();
+        DocumentReference queueSizeReference =
+            new DocumentReference("QueueSize", testReference.getLastSpaceReference());
+        setup.createPage(queueSizeReference,
+            "{{velocity}}$services.eventstream.getQueueSize(){{/velocity}}");
+
+        setup.getDriver().waitUntilCondition(input -> {
+            setup.gotoPage(queueSizeReference, "get", "outputSyntax=plain&forceeload=" + System.currentTimeMillis());
+            String content = setup.getDriver().findElement(By.tagName("body")).getText();
+            return content.equals("0");
+        });
+
```

Comment:
That's not great... ideally I would have replaced the `NotificationsTrayPage.waitOnNotificationCount` with this new way for waiting on the empty queue